output/output.go

```diff
@@ -1,24 +1,106 @@
+// command will return just the raw public key. Even though there are other fields
+// returned in the response body, the public key is almost always what the user
+// wants from the command and printing out just the public key makes it easy to
+// copy (and paste).
+//
+// This function accepts 3 parameters:
+// 1) a pointer to the cli.Context: this is for checking the flags to determine the
+// 		output format.
+// 2) a pointer to a function that returns an `interface{}`: this (pointer to a)
+// 		function returns the data from the response body that gets returned from the
+//		command request. It returns an `interface{}` to accomodate the different data
+//		types that can be printed (e.g. `map[string]interface{}`, `[]map[string]interface{}`,
+//		`string`, etc). Some commands can print out the data that gets returned as-is,
+//		for example if all the values are `string` or `int`. In these cases, a simple
+//		operation that converts the value(s) from the response body into `map[string]interface{}`
+//		(or `[]map[string]interface{}`) is sufficient. In such cases, the parameter `f`
+//		in the `Print` function might look like this:
+//
+//		f := func() interface{} {
+//			// o would be defined above and is the raw (or type casted) value
+//			// returned in the response body.
+//			// This function returns an `interface{}` that has a `map[string]interface{}`
+//			// type.
+//			return structs.Map(o)
+//		}
+//
+//		However, in some cases it is necessary to massage the data to get it into
+//		a presentable form, like if the response body contains nested maps and one
+//		of the nested maps has a field that you want to output. Below is an example
+//		of a function for a `list` command:
+//
+//		f := func() interface{} {
+//			// o would be defined above and is the raw (or type casted) value
+//			// returned in the response body
+//			m := make([]map[string]interface{}, len(o))
+//			for j, kp := range o {
+//				m[j] = structs.Map(kp)
+//			}
+//			// This function returns an `interface{}` that has a `[]map[string]interface{}`
+//			// type.
+//			return m
+//		}
+//
+//		Still yet, there may be cases when you just want a single piece of raw data
+//		(like in the case of `rack servers keypair get`). In these cases, you can
+//		choose to return a single field from the response body as a string:
+//
+//			f := func() interface{} {
+//				// o would be defined above and is the raw (or type casted) value
+//				// returned in the response body
+//				m := structs.Map(o)
+//				// This function returns an `interface{}` that has a `string` type.
+//				return m["PublicKey"]
+//			}
+//
+//		Regardless of how the function looks, it is created within the command
+// 		function as a closure around the data from response body.
+//
+// 3) a slice of strings: this slice contains the header values to print out for
+// 		the tabular and csv formats.
+func Print(c *cli.Context, f *func() interface{}, keys []string) {
```

Comment:
Looks good, thank you so much for clearing this up for contributors. If I changed anything, I'd turn `1)` -> `c`, `2)` -> `f`, and `3)` -> `keys` so they match the param name.